src/gui/sdlmain.cpp

```diff
@@ -1818,6 +1816,49 @@ static void DisplaySplash(uint32_t time_ms)
 	SDL_Delay(time_ms);
 }
 
+/* For some preference values, we can safely remove comment after # character
+ */
+static std::string NormalizeConfValue(const char *val)
+{
+	std::string pref = val;
+	const auto comment_pos = pref.find('#');
+	if (comment_pos != std::string::npos)
+		pref.erase(comment_pos);
+	trim(pref);
+	lowcase(pref);
+	return pref;
+}
+
+static void SetupWindowResolution(const char *val)
+{
+	assert(sdl.display_number >= 0);
+	std::string pref = NormalizeConfValue(val);
+
+	if (pref == "original" || pref.empty()) {
+		sdl.desktop.window.use_original_size = true;
+		return;
+	}
+
+	int w = 0;
+	int h = 0;
+	// sscanf won't parse suffix after second integer
+	if (sscanf(pref.c_str(), "%dx%d", &w, &h) == 2) {
+		SDL_Rect bounds;
+		SDL_GetDisplayBounds(sdl.display_number, &bounds);
+		if (w > 0 && h > 0 && w <= bounds.w && h <= bounds.h) {
+			sdl.desktop.window.use_original_size = false;
+			sdl.desktop.window.width = w;
+			sdl.desktop.window.height = h;
```

Comment:
Also thumbs up to simplifying the parsing and ditching the percent handling!

I think Window-mode sizing and customization is an area where fewer options is better for users (and the code).

For example: `original` is more harmful than good, given it produces a completely unplayable 320x200 postage-stamp for the majority of DOS games.  Even on a bog-standard 1996 17" CRT 1280x1024 monitor, 320x200 is unplayable.. so this is where we could eliminate "original" and use 640x480 as a minimum default bound; or even default to the first-pixel-perfect increment.